Album/ViewControllers/ProfileControllers/SettingsViewController.m

```diff
@@ -5,38 +5,205 @@
+                self.profileImageView.layer.masksToBounds = YES;
+            }
+        }];
+    }
+    
+    // Display current user marker color
+    if (user[@"colorHexString"]) {
+        UIColor *color = [self.colorHelper colorFromHexString:user[@"colorHexString"]];
+        [self.colorView setImage:[self.colorHelper createImageWithColor:color]];
+    }
+    self.usernameField.text = user.username;
+    self.emailField.text = user.email;
+    self.pwField.text = user.password;
+    self.pwField.placeholder = @"Please enter non-empty password";
+} /* setCurrentView */
+
+- (void)didTapUserProfile:(UITapGestureRecognizer *)sender {
+    if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
+        UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Media" message:@"Choose"
+                                                                preferredStyle:(UIAlertControllerStyleAlert)];
+        // Take photo action
+        UIAlertAction *photoAction = [UIAlertAction actionWithTitle:@"Take Photo"
+                                                              style:UIAlertActionStyleCancel
+                                                            handler:^(UIAlertAction *_Nonnull action) {
+            UIImagePickerController *imagePickerVC = [UIImagePickerController new];
+            imagePickerVC.delegate = self;
+            imagePickerVC.allowsEditing = YES;
+            imagePickerVC.sourceType = UIImagePickerControllerSourceTypeCamera;
+            [self presentViewController:imagePickerVC animated:YES completion:nil];
+        }];
+        // Add the take photo action to the alertController
+        [alert addAction:photoAction];
+        // Create an upload from library action
+        UIAlertAction *uploadAction = [UIAlertAction actionWithTitle:@"Upload from Library"
+                                                               style:UIAlertActionStyleDefault
+                                                             handler:^(UIAlertAction *_Nonnull action) {
+            UIImagePickerController *imagePickerVC = [UIImagePickerController new];
+            imagePickerVC.delegate = self;
+            imagePickerVC.allowsEditing = YES;
+            imagePickerVC.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+            [self presentViewController:imagePickerVC animated:YES completion:nil];
+        }];
+        // Add the upload from library action to the alert controller
+        [alert addAction:uploadAction];
+        //Cancel
+        UIAlertAction *cancel = [UIAlertAction actionWithTitle:@"Cancel"
+                                                         style:UIAlertActionStyleDefault
+                                                       handler:nil];
+        // Add the cancel action to the alert controller
+        [alert addAction:cancel];
+        [self presentViewController:alert animated:YES completion:nil];
```

Comment:
LEts move the alerts to a helper function

Reply:
Done!